rita/src/rita_client/exit_manager/mod.rs

```diff
@@ -269,11 +272,23 @@ impl Handler<Tick> for ExitManager {
             trace!("We have selected an exit!");
             if let Some(ref general_details) = exit.info.general_details() {
                 trace!("We have details for the selected exit!");
-                if let Some(_) = exit.info.our_details() {
+                // only run if we have our own details and we either have no setup exit or the chosen
+                // exit has changed
+                if exit.info.our_details().is_some()
+                    && !(self.last_exit.is_some() && self.last_exit.clone().unwrap() == exit)
+                {
+                    // fully cleanup exit tunnel before making a change
+                    match KI.del_interface("wg_exit") {
```

Comment:
If you mean for a full cleanup, don't you need to also remove the default route?

Reply:
true I suppose you could modfiy the gateway ip.